examples/misc/zorder_demo.py

```diff
@@ -3,66 +3,69 @@
 Zorder Demo
 ===========
 
-The default drawing order for axes is patches, lines, text.  This
-order is determined by the zorder attribute.  The following defaults
-are set
+The drawing order of artists is determined by their ``zorder`` attribute, which
+is a floating point number. Artists with higher ``zorder`` are drawn on top.
+You can change the order for individual artists by setting their ``zorder``.
+The default value depends on the type of the Artist:
 
-=======================    =======
-Artist                     Z-order
-=======================    =======
-Patch / PatchCollection    1
-Line2D / LineCollection    2
-Text                       3
-=======================    =======
+================================================================    =======
+Artist                                                              Z-order
+================================================================    =======
+Images (`.AxesImage`, `.FigureImage`, `.BboxImage`)                 0
+`.Patch`, `.PatchCollection`                                        1
+`.Line2D`, `.LineCollection` (including minor ticks, grid lines)    2
+Major ticks                                                         2.01
+`.Text` (including axes labels and titles)                          3
+`.Legend`                                                           5
```

Comment:
Many of these classes are a bit obscure to users.  Can we include the `Axes` calls that make them?

Reply:
I agree that these classes are a bit obscure. OTOH, the Z-order is exactly attached to these classes. It does not match too well to `Axes` calls, at least I found no consistent and descriptive way: `Line2D` can be created by a number of methods: `plot` (and all of its derivatives like `semilogx`, ...), `axh/vline`, `scatter` (depending on the parameters), ... Major ticks are implicitly made, there's no `Axes` method that create them. Even though `Text` and `Rectangle` are also building blocks for tables, I would not reference `table()` as a patch-creating method.

Therefore, I sticked to the existing structure of the table and just added more entries. Ideas to improve on that are always welcome.

Reply:
I would just do `e.g. ax.plot, including minor...` without trying to be comprehensive...

Reply:
Feel free to push an update if you like. I'm not feeling like fiddling what commands should get added right now.

Anyway, IMHO this PR stands for for itself and could get merged as an incremental improvement as is. Further improvements are always possible later.